statsmodels/survey/summary_stats.py

```diff
@@ -0,0 +1,612 @@
+"""
+Methods for creating summary statistics and their SE for survey data.
+
+The main classes are:
+
+  * SurveyDesign : Parent class that creates attributes for easy
+  implementation of other methods. Attributes include relabeled
+  clusters, number of clusters per strata, etc.
+
+  * SurveyStat : implements methods to calculate the standard
+  error of each statistic via either the bootstrap or jackknife
+
+  * SurveyMean : Calculates the mean of each column
+
+  * SurveyTotal : Calculates the total of each column
+
+  * SurveyQuantile: Calculates the specified quantile[s] of each column
+"""
+
+import numpy as np
+# import pandas as pd
+
+
+class SurveyDesign(object):
+    """
+    Description of a survey design, used by most methods
+    implemented in this module.
+
+    Parameters
+    -------
+    strata : array-like or None
+        Strata for each observation. If none, an array
+        of ones is constructed
+    cluster : array-like or None
+        Cluster for each observation. If none, an array
+        of ones is constructed
+    weights : array-like or None
+        The weight for each observation. If none, an array
+        of ones is constructed
+    nest : boolean
+        allows user to specify if PSU's with the same
+        PSU number in different strata are treated as distinct PSUs.
+
+    Attributes
+    ----------
+    weights : (n, ) array
+        The weight for each observation
+    nstrat : integer
+        The number of district strata
+    sclust : (n, ) array
+        The relabeled cluster array from 0, 1, ..
+    strat : (n, ) array
+        The related strata array from 0, 1, ...
+    ncs : (self.nstrat, ) array
+        Holds the number of clusters in each stratum
+    sfclust : ndarray
+        The stratum for each cluster
+    nclust : integer
+        The total number of clusters across strata
+    """
+
+    def __init__(self, strata=None, cluster=None, weights=None,
```

Comment:
I'm not sure about that. I would leave it for now as `cov_method`, because it will be easy to do a global replace later.
The main issue that I'm not sure about is that we might need both. if cov_type specifies the correlation or heteroscedasticity structure, then for each structure we could use bootstrap or jacknife, 
for example using wild bootstrap for heteroscedasticity could be either `cov_type='hc_boots'` or `cov_type='HC', cov_method='boots'